superset/models/helpers.py

```diff
@@ -1974,7 +1974,9 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                 and db_engine_spec.allows_hidden_cc_in_orderby
                 and col.name in [select_col.name for select_col in select_exprs]
             ):
-                col = literal_column(col.name)
+                engine = self.database._get_sqla_engine()
+                quote = engine.dialect.identifier_preparer.quote
+                col = literal_column(quote(col.name))
```

Comment:
Interesting. Typically the SQLAlchemy dialect should handle the quoting. Additionally it's unclear whether forceably quoting the column would be problematic when combined with an alias.

@villebro do you have any insights on this? I see previously you registered https://github.com/sqlalchemy/sqlalchemy/issues/4730 related to SQLAlchemy not quoting.